File: Goods.Social.Network/Goods.System.Social.Network.Microservice.Users/Entities/DTO/UserDTO.cs
namespace Goods.System.Social.Network.Microservice.Users.Entities.DTO
{
    /// <summary>
    /// Модель пользователя DTO
    /// </summary>
    public class UserDTO
    {
        /// <summary>
        /// Id юзера
        /// </summary>
        public int Id { get; set; }

        /// <summary>
        /// Имя
        /// </summary>
        public string FirstName { get; set; } = string.Empty;

        /// <summary>
        /// Фамилия
        /// </summary>
        public string LastName { get; set; } = string.Empty;

        /// <summary>
        /// Дата рождения
        /// </summary>
        public DateTime DateBirthday { get; set; }

        /// <summary>
        /// Город
        /// </summary>
        public string City { get; set; } = string.Empty;

        /// <summary>
        /// Страна
        /// </summary>
        public string Country { get; set; } = string.Empty;

        /// <summary>
        /// Статус
        /// </summary>
        public string Status { get; set; } = string.Empty;

        /// <summary>
        /// Электронная почта
        /// </summary>
        public string Email { get; set; } = string.Empty;

        /// <summary>
        /// Подписан ли твой юзера на этого
        /// </summary>
        public bool IsSubscriber { get; set; }

        /// <summary>
        /// Путь до аватара
        /// </summary>
        public string Avatar { get; set; } = "0.jpeg";

        /// <summary>
        /// Путь до фона
        /// </summary>
        public string Background { get; set; } = "0.jpeg";
    }
}

File: web-app/libs/data/src/controllers/user/user-dto.ts
import { WhoEnum } from '../../models';

/** PUT /api/user Body */
export type UserPutBody = {
  id: number;
  email: string | null;
  firstName: string | null;
  lastName: string | null;
  city: string | null;
  country: string | null;
  dateBirthday: Date | string | null;
  status: string | null;
  phone: string | null;
  avatar: string | null;
  background: string | null;
  // Новый JWT токен
};

/** GET /api/user/{id} Return */
export type UserGetReturn = {
  id: number;
  firstName: string | null;
  lastName: string | null;
  dateBirthday: Date | string;
  city: string | null;
  country: string | null;
  status: string | null;
  email: string | null;
  isSubscriber: boolean;
  avatar: string | null;
  background: string | null;
};

/** GET /api/user Query */
export type UserGetListQuery = {
  search: string;
  number: number;
  who: WhoEnum;
};

/** GET /api/user Return */
export type UserGetListReturn = {
  countAllUsers: number;
  pageCount: number;
  numberPage: number;
  users: UserGetReturn[];
};
